### DISCUSSION TITLE
Custom Notification/Alert dialogs or 3rd party libs?

### DISCUSSION THREAD

**Discussion by mathiasayivor** [ykdojo/defaang#176] [Ideas]
Are there any plans for using 3rd party libraries like [Sweet Alert](https://sweetalert2.github.io/) or Custom solutions built with tailwind for notifications/alerts/dialogs?

I'm thinking of adding a custom solution using tailwind, which we can use to give feedback to users (alerts, dialogs, etc).
The API would be in the form of:
```
import {useFeedback, Modal} from '@/lib/feedback'
const {notify, dialog} = useFeedback()
const {showModal, setShowModal} = useState(false)

function someHandler(...args) {
    // Do some computing here
    const color = 'primary'
    notify('Message', color).onDismissed(() => {
        console.log('Notification Dismissed')
    })

    dialog({
        title: 'Title here',
        message: 'Proceed with action?',
        confirmText: 'Yes',
        cancelText: 'Cancel'
    }).onResponse(response => {
        if (response) {
            console.log('User agreed')
        } else {
            console.log('User declined')
        }
    }).onDismissed(() => {
        console.log('Dialog dismissed')
    }).onHide(() =>{
        console.log('callback after dialog is hidden after any action is dispatched')
    })
}

return (
<Modal show={showModal}>
    // Dialog Content here...
</Modal>
)
```
This makes it easier to give feedback to users after actions on the platform + customization options to match the theme of the app

**Comment by CerealPlayer** [comment]
We could definitely use some notifications solution. What do you think of adding some UI library like Mantine or MUI? They come with notifications and with some more stuff we could use across the app

**Reply by Gulshanaggarwal** [reply]
I think yes, because alerts are now boring, we need something fancy nowadays

**Reply by ykdojo** [reply]
For UI, we're using Tailwind UI currently
